cli/kubectl_kadalu/storage_remove.py

```diff
@@ -0,0 +1,192 @@
+"""
+'storage-remove' sub command
+"""
+
+# noqa # pylint: disable=duplicate-code
+# noqa # pylint: disable=too-many-branches
+
+#To prevent Py2 to interpreting print(val) as a tuple.
+from __future__ import print_function
+
+import os
+import tempfile
+import sys
+import json
+
+import utils
+from storage_yaml import to_storage_yaml
+
+
+def set_args(name, subparsers):
+    """ add arguments, and their options """
+
+    parser = subparsers.add_parser(name)
+    arg = parser.add_argument
+
+    arg(
+        "name",
+        help="Storage Name"
+    )
+    utils.add_global_flags(parser)
+
+
+def validate(args):
+    """
+    Validate the storage requested to be deleted
+    is present in kadalu configmap or not.
+    Exit if not present.
+    """
+
+    storage_info_data = get_configmap_data(args.name)
+
+    if storage_info_data is None:
+        print("Aborting.....")
+        print("Invalid name. No such storage '%s' in Kadalu configmap." % args.name)
+        sys.exit(1)
+
+
+def get_configmap_data(volname):
+    """
+    Get storage info data from kadalu configmap
+    """
+
+    cmd = ["sudo", "kubectl", "get", "configmap", "kadalu-info", "-nkadalu", "-ojson"]
+
+    try:
+        resp = utils.execute(cmd)
+        config_data = json.loads(resp.stdout)
+
+        data = config_data['data']
+        storage_name = "%s.info" % volname
+        storage_info_data = data[storage_name]
+
+        # Return data in 'dict' format
+        return json.loads(storage_info_data)
+
+    except utils.CommandError as err:
+        utils.command_error(cmd, err.stderr)
+
+    except KeyError:
+        # Validate method expects None when 'storage' not found.
+        return None
+
+
+def storage_add_data(args):
+    """ Build the config file """
+
+    storage_info_data = get_configmap_data(args.name)
+
+    content = {
+        "apiVersion": "kadalu-operator.storage/v1alpha1",
+        "kind": "KadaluStorage",
+        "metadata": {
+            "name": args.name
+        },
+        "spec": {
+            "type": storage_info_data['type'],
+            "storage": []
+        }
+    }
+
+    # External details are specified, no 'storage' section required
+    if storage_info_data['type'] == "External":
+        # TODO: Check keynames from kadalu configmap and set node, vol
```

Comment:
@amarts I'm not sure of their keynames in kadalu configmap. Is it node = brick['node']. And for vol?

Reply:
Do without this section altogether.